TPM simulator errors on Macbook M1

<!--
If you are looking for help, please join us on [Slack or the mailing
lists.](https://github.com/spiffe/spiffe/blob/master/README.md)

If you are submitting a feature request or bug report, please be
as detailed as possible.
-->

* **Version**: all
* **Platform**: Mac m1
* **Subsystem**: Makefile system

The M1 mac has an ARM based processor. The linter has some unexpected complaints about the TPM simulator:

```
✗ make lint
test/tpmsimulator/simulator.go:82:33: cannot use sim (variable of type *TPMSimulator) as io.ReadWriter value in argument to tpm2.HierarchyChangeAuth: missing method Read (typecheck)
	err = tpm2.HierarchyChangeAuth(sim,
	                               ^
test/tpmsimulator/simulator.go:100:33: cannot use sim (variable of type *TPMSimulator) as io.ReadWriter value in argument to tpm2.HierarchyChangeAuth: missing method Read (typecheck)
	err = tpm2.HierarchyChangeAuth(sim,
	                               ^
test/tpmsimulator/simulator.go:209:28: cannot use s (variable of type *TPMSimulator) as io.Reader value in argument to io.NopCloser: missing method Read (typecheck)
		ReadCloser: io.NopCloser(s),
		                         ^
test/tpmsimulator/simulator.go:210:15: cannot use s (variable of type *TPMSimulator) as io.Writer value in struct literal: missing method Write (typecheck)
		Writer:     s,
		            ^
```

Test failures, so far. Might be because of my openssl version:

```
ld: library not found for -lcrypto
clang: error: linker command failed with exit code 1 (use -v to see invocation)
# github.com/google/go-tpm-tools/simulator/internal
In file included from include.c:153:
/Users/fmemon/go/pkg/mod/github.com/google/go-tpm-tools@v0.3.4/simulator/ms-tpm-20-ref/TPMCmd/tpm/src/crypt/CryptCmac.c:104:5: warning: 'AES_set_encrypt_key' is deprecated [-Wdeprecated-declarations]
/Users/fmemon/go/pkg/mod/github.com/google/go-tpm-tools@v0.3.4/simulator/ms-tpm-20-ref/TPMCmd/tpm/include/CryptSym.h:134:9: note: expanded from macro 'SELECT'
<scratch space>:163:1: note: expanded from here
/Users/fmemon/go/pkg/mod/github.com/google/go-tpm-tools@v0.3.4/simulator/ms-tpm-20-ref/TPMCmd/tpm/include/CryptSym.h:102:29: note: expanded from macro 'ENCRYPT_CASE_AES'
/Users/fmemon/go/pkg/mod/github.com/google/go-tpm-tools@v0.3.4/simulator/ms-tpm-20-ref/TPMCmd/tpm/include/CryptSym.h:92:9: note: expanded from macro 'ENCRYPT_CASE'
```

--- a/pkg/agent/plugin/nodeattestor/tpmdevid/devid_test.go
+++ b/pkg/agent/plugin/nodeattestor/tpmdevid/devid_test.go
@@ -1,3 +1,6 @@
+//go:build !darwin
+// +build !darwin
+
 package tpmdevid_test
 
 import (

--- a/pkg/agent/plugin/nodeattestor/tpmdevid/tpmutil/session_test.go
+++ b/pkg/agent/plugin/nodeattestor/tpmdevid/tpmutil/session_test.go
@@ -1,3 +1,6 @@
+//go:build !darwin
+// +build !darwin
+
 package tpmutil_test
 
 import (

--- a/pkg/server/plugin/nodeattestor/tpmdevid/devid_test.go
+++ b/pkg/server/plugin/nodeattestor/tpmdevid/devid_test.go
@@ -1,3 +1,6 @@
+//go:build !darwin
+// +build !darwin
+
 package tpmdevid_test
 
 import (

--- a/test/tpmsimulator/simulator.go
+++ b/test/tpmsimulator/simulator.go
@@ -1,3 +1,6 @@
+//go:build !darwin
+// +build !darwin
+
 package tpmsimulator
 
 import (

